fix ddp logging test

zhaojuanmao opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#56640 fix ddp logging test**

reset performance stats for current iteration, also fix ddp logging verifiction for sampled iterations.

Differential Revision: [D27923414](https://our.internmc.facebook.com/intern/diff/D27923414/)

zhaojuanmao committed:
fix ddp logging test

reset performance stats for current iteration, also fix ddp logging verifiction for sampled iterations.

Differential Revision: [D27923414](https://our.internmc.facebook.com/intern/diff/D27923414/)

[ghstack-poisoned]

SciPioneer commented on torch/lib/c10d/logger.cpp:
```diff
@@ -204,6 +204,12 @@ void Logger::set_runtime_stats_and_log() {
     ddp_logging_data_->rebuilt_bucket_sizes = get_bucket_sizes();
   }
 
+  // reset perforamnce stats for current iteration.
```
Nit: It will be better to wrap up these lines a as `reset_perf_stats` function, and you can put anything that needs to be reset per iteration in it in the future.

zhaojuanmao committed:
Update on "fix ddp logging test"


reset performance stats for current iteration, also fix ddp logging verifiction for sampled iterations.

Differential Revision: [D27923414](https://our.internmc.facebook.com/intern/diff/D27923414/)

[ghstack-poisoned]
